src/TMS/CourseActionImpl.php

```diff
@@ -31,21 +31,25 @@
 	 */
 	protected $contexts;
 
+	/**
+	 * @var int
+	 */
+	protected $current_user_id;
+
 	/**
 	 * @param Ente\Entity 	$entity
 	 * @param \ilObject 	$owner
 	 * @param int 	$priority
 	 * @param int[] 	$contexts
 	 */
-	public function __construct(Ente\Entity $entity, \ilObject $owner, $priority, array $contexts) {
+	public function __construct(Ente\Entity $entity, \ilObject $owner, \ilObjUser $current_user, $priority, array $contexts) {
```

Comment:
Wieso den ganzen user, wenn wir davon nur die ID brauchen? Ist current user == global user?

Reply:
Ja, ich hab auf diese Art da eine versteckte Dependency auf `$DIC->user()` entfernt. Im Grunde ist dass hier einfach eine billige Übersetzung, ohne groß drüber nachzudenken. Die Frage ist also berechtigt. Pro wäre: Ich weiß so innen, das ich bestimmt die ID eines Users hab. Contra ist, dass wir das nirgendwo anders so machen.

Reply:
Na wer weiß was wir damit demnächst machen sollen/wollen. Von daher finde ich es nicht so schlimm. Vielleicht nur noch in den Docstring rein, das es der global user ist. Dann ist die Frage auch geklärt.

Reply:
done